src/selection-match.ts

```diff
@@ -59,27 +74,35 @@ const matchHighlighter = ViewPlugin.fromClass(class {
     let conf = view.state.facet(highlightConfig)
     let {state} = view, sel = state.selection
     if (sel.ranges.length > 1) return Decoration.none
-    let range = sel.main, query, check = null
+    let range = sel.main, query, check = null, empty = false
     if (range.empty) {
       if (!conf.highlightWordAroundCursor) return Decoration.none
       let word = state.wordAt(range.head)
       if (!word) return Decoration.none
+      empty = true
       check = state.charCategorizer(range.head)
       query = state.sliceDoc(word.from, word.to)
     } else {
       let len = range.to - range.from
       if (len < conf.minSelectionLength || len > 200) return Decoration.none
-      query = state.sliceDoc(range.from, range.to).trim()
-      if (!query) return Decoration.none
+      if (conf.wholeWords) {
+        query = state.sliceDoc(range.from, range.to) // TODO: allow and include leading/trailing space?
+        if (!query) return Decoration.none
```

Comment:
How could slicing a non-empty range ever provide a falsy value?

Reply:
Good point! In the existing code there's an extra ` .trim()` which could lead to a falsy value, but it's not present here so this check isn't necessary.